File: Consolidator.Max/js/ViewModels/FilterCatalog.js
const EqualizerFilterDefinitions = [
    { type: "gain", fixedQ: 0.707, parameters: {
        gain: { minimum: -24, maximum: 24, defaultValue: 0 }
    } },
    { type: "tilt", fixedQ: 0.707, parameters: {
        frequency: { minimum: 20, maximum: 20000, defaultValue: 1000 },
        gain: { minimum: -24, maximum: 24, defaultValue: 0 }
    } },
    { type: "low_shelf", fixedQ: 0.707, parameters: {
        frequency: { minimum: 20, maximum: 20000, defaultValue: 100 },
        gain: { minimum: -24, maximum: 24, defaultValue: 0 }
    } },
    { type: "high_shelf", fixedQ: 0.707, parameters: {
        frequency: { minimum: 20, maximum: 20000, defaultValue: 10000 },
        gain: { minimum: -24, maximum: 24, defaultValue: 0 }
    } },
    { type: "bell", fixedQ: 0.707, parameters: {
        frequency: { minimum: 20, maximum: 20000, defaultValue: 1000 },
        q: { minimum: 0.1, maximum: 10, defaultValue: 0.707 },
        gain: { minimum: -24, maximum: 24, defaultValue: 0 }
    } },
    { type: "bell", fixedQ: 0.707, parameters: {
        frequency: { minimum: 20, maximum: 20000, defaultValue: 2000 },
        q: { minimum: 0.1, maximum: 10, defaultValue: 0.707 },
        gain: { minimum: -24, maximum: 24, defaultValue: 0 }
    } },
    { type: "bell", fixedQ: 0.707, parameters: {
        frequency: { minimum: 20, maximum: 20000, defaultValue: 4000 },
        q: { minimum: 0.1, maximum: 10, defaultValue: 0.707 },
        gain: { minimum: -24, maximum: 24, defaultValue: 0 }
    } }
];

const DetectorFilterDefinitions = [
    { type: "low_shelf", fixedQ: 0.707, parameters: {
        frequency: { minimum: 20, maximum: 20000, defaultValue: 100 },
        gain: { minimum: -24, maximum: 24, defaultValue: 0 }
    } },
    { type: "bell", fixedQ: 0.707, parameters: {
        frequency: { minimum: 20, maximum: 20000, defaultValue: 1000 },
        q: { minimum: 0.1, maximum: 10, defaultValue: 0.707 },
        gain: { minimum: -24, maximum: 24, defaultValue: 0 }
    } }
];

module.exports = {
    EqualizerFilterDefinitions: EqualizerFilterDefinitions,
    DetectorFilterDefinitions: DetectorFilterDefinitions
};

File: Consolidator.Managed/Core/Settings/FilterCatalog.cs
using Consolidator.Managed.Core.State;

namespace Consolidator.Managed.Core.Settings;

public enum FilterKind
{
    Bell,
    LowShelf,
    HighShelf,
    Tilt,
    Gain
}

public sealed record FilterParameterDefinition(
    float DefaultValue,
    FloatRange Range);

public sealed record FilterDefinition(
    FilterKind Kind,
    FilterParameterDefinition? Frequency,
    FilterParameterDefinition? Q,
    FilterParameterDefinition Gain,
    float FixedQ);

public static class FilterCatalog
{
    public static readonly FilterDefinition[] Equalizer =
    [
        Create(FilterKind.Gain, null, null, 0.0F, 0.707F),
        Create(FilterKind.Tilt, 1000.0F, null, 0.0F, 0.707F),
        Create(FilterKind.LowShelf, 100.0F, null, 0.0F, 0.707F),
        Create(FilterKind.HighShelf, 10000.0F, null, 0.0F, 0.707F),
        Create(FilterKind.Bell, 1000.0F, 0.707F, 0.0F, 0.707F),
        Create(FilterKind.Bell, 2000.0F, 0.707F, 0.0F, 0.707F),
        Create(FilterKind.Bell, 4000.0F, 0.707F, 0.0F, 0.707F)
    ];

    public static readonly FilterDefinition[] Detector =
    [
        Create(FilterKind.LowShelf, 100.0F, null, 0.0F, 0.707F),
        Create(FilterKind.Bell, 1000.0F, 0.707F, 0.0F, 0.707F)
    ];

    public static string ToProtocolName(FilterKind kind) => kind switch
    {
        FilterKind.Bell => "bell",
        FilterKind.LowShelf => "low_shelf",
        FilterKind.HighShelf => "high_shelf",
        FilterKind.Tilt => "tilt",
        FilterKind.Gain => "gain",
        _ => throw new ArgumentOutOfRangeException(nameof(kind))
    };

    public static IReadOnlyList<FilterDefinition> For(ProcessorId processorId) =>
        processorId switch
        {
            ProcessorId.Equalizer => Equalizer,
            ProcessorId.Saturator or ProcessorId.Compressor => Detector,
            _ => Array.Empty<FilterDefinition>()
        };

    private static FilterDefinition Create(
        FilterKind kind,
        float? frequency,
        float? q,
        float gain,
        float fixedQ)
    {
        return new FilterDefinition(
            kind,
            CreateParameter(frequency, DspParameterRanges.FrequencyHz),
            CreateParameter(q, DspParameterRanges.Q),
            new FilterParameterDefinition(gain, DspParameterRanges.FilterGainDb),
            fixedQ);
    }

    private static FilterParameterDefinition? CreateParameter(
        float? defaultValue,
        FloatRange range)
    {
        return defaultValue is { } value
            ? new FilterParameterDefinition(value, range)
            : null;
    }
}
